### ISSUE TITLE
Pangeo Setup Guides for Beginners

### ISSUE THREAD

**Issue by chiaral** [pangeo-data/pangeo#108]
I have never worked on a machine like Cheyenne, so my learning curve is pretty steep. I don't think Pangeo setup guides should worry about explaining everything, but I think that together with an "[For Advanced Users](https://pangeo-data.github.io/pangeo/setup_guides/index.html#)" guide we need a "For Beginner Users" guide . 

I followed steps 1 and 2 in the guide, but at the moment I don't want to launch a big job using Dask. I just want to launch Jupyter in interactive mode and browse GLADE. 

On Cheyenne guide pages they have this explanation on how to launch [iPython or Jupyter ](https://www2.cisl.ucar.edu/resources/computational-systems/cheyenne/software/jupyter-and-ipython) in interactive mode. They run qinteractive, load the cisl python module, ncar_pylib and use a "start_notebook" command. 

Running start_notebook after I activate pangeo (I used the environment.yml file in our repo to create this environment) doesn't work of course. 

Can we have a launch_notebook.sh similarly to the launch_dask.sh file in our repo (that works with the local miniconda python) and launch only jupyter? (and I know that I am vague - because I am not sure what else should be included). 

Maybe the answer to this question is very simple and already in matt's guide, just not quite clear to me - and people like me who know nothing about this stuff. If you make it clearer I am happy to include it in a document I started working on, that could eventually become a "For Beginners Guide". So far it has Matt's steps 1 and 2 (plus few things that were not necessarily obvious to me) and I will add more as I make my way through it.

**Comment by jhamman** [comment]
@chiaral - thank you for outlining these issues in detail. In some ways, I was expecting this to come from someone soon based on some of the issues described in #94. 

I think there are two ways we can approach this:

#### Use the system install of Python/Jupyter and install a Pangeo Jupyter Kernel

This has the advantage of being fairly light weight but requires users to install their own jupyter kernel. Another disadvantage is that things like #94 can pop up and well, you might just be left to sort it out on your own anyways.

The workflow for this approach would be:

```
# log into cheyenne
ssh ...

# install the pangeo kernel spec
# this will let jupyter know where to find the pangeo environment
# it only has to be done once and can be done for many environments
source activate pangeo
python -m ipykernel install --user --name pangeo --display-name "Python (pangeo)"

# follow instructions for getting an interactive node
qinteractive -l walltime=6:00:00

# load modules
module load python jupyter

# start a notebook
start-notebook 

# follow the instructions that this command produces
# once inside jupyter, select the "Python (pangeo)" kernel from the kernels drop down menu
```

#### Use your own environment to launch jupyter

This has the advantage of being considerably more flexible and customizable. For example, maybe you want to use jupyter lab instead of jupyter notebook. The downside is you are going to have to launch the jupyter notebook yourself.

The workflow for this approach would be:

```
# log into cheyenne
ssh ...

# follow instructions for getting an interactive node
qinteractive -l walltime=6:00:00

# activate pangeo (or another environment)
source activate pangeo

# start a notebook server
start-pangeo-notebook 

# follow the instructions that this command produces
```

This approach could also be combined with the above approach wherein multiple kernels can still be installed.

I have already created something akin to `start-pangeo-notebook` and I would be happy to put it up here. Before I do that though, I'm curious how these two workflows compare in terms of their perceived barriers to entry, management concerns, etc. Obviously, I have already moved to the second option but don't want to advocate for that if its more involved than folks think is necessary.

**Comment by rabernat** [comment]
I think we make an effort to work with the tools CISL provides as much as possible. They have already put in effort to create a notebook launching script, and this may soon be followed by a full-blown jupyter service. It would be nice if pangeo didn't have to support / maintain / document a workaround for this.

If we could simplify / document the process for creating user-defined kernels, then we can just focus on that part. Together with `PBSCluster`, I think this would allow us to avoid maintaining all of these somewhat duplicative / buggy launch scripts.

**Comment by chiaral** [comment]
Thanks for the answers. I am fine either way - as long as I have instructions to follow!

@jhamman is the first worflow complete? Do I simply install the kernel and then select it in the notebook?

`module load python jupyter`

loads python 2.7, so I assume that installing a pangeo-based kernel will overwrite that?

in fact 

`module load python/3.6.4 jupyter `

won't work, because the module jupyter requires python/2.7.13

**Comment by jhamman** [comment]
Yes, it should be complete. Python 2.7 will be used to run the Jupyter server but your environment (kernel) can be python 3. I'm curious if you tried this as is and if it worked for you.

**Comment by chiaral** [comment]
Ok I managed to get it working but not following your instructions...

First I followed your instructions and opened the notebook successfully
(FYI once I loaded the python module I've got the message 
```
clepore@r8i4n0:~> module load python jupyter

The following have been reloaded with a version change:
  1) python/3.6.4 => python/2.7.13

```
also the jupyter psswd I set up following matt's instruction did not work, so I used the token)

I can select the kernel, but then it dies immediately and it cannot be restarted.

I tried doing it following your instructions completely, you do not deactivate pangeo.

I instead modified your instructions as follows:

```
# log into cheyenne
ssh ...

# I installed the pangeo kernel spec once
source activate pangeo
python -m ipykernel install --user --name pangeo --display-name "Python (pangeo)"

#I deactivate pangeo
source deactivate

# I get the node
qinteractive -l walltime=6:00:00

# I load python 3
module load python/3.6.4

#I activate their package library
ncar_pylib
# this is the only way I figured out to make start-notebook work with cisl python 3..

# start a notebook
start-notebook 

# follow the instructions that this command produces
# once inside jupyter, select the "Python (pangeo)" kernel from the kernels drop down menu
# it seems to work
```
I should add that if I followed your instruction, the python 2 kernel had no package in it.. I guess because no environment was loaded for that kernel.

I tried to be careful - but it might be that i made some mistake.

**Comment by jhamman** [comment]
@chiaral - I'm happy to chat with you early this week to help you get going on Cheyenne if you think that would help.

**Comment by chiaral** [comment]
@jhamman thanks a lot. you can message me a few times/days that work for you, and I will be there.
